src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1535,8 +1535,11 @@ void TorrentImpl::endReceivedMetadataHandling(const Path &savePath, const PathLi
         m_filePaths.append(filePath);
 
         lt::download_priority_t &nativePriority = p.file_priorities[LT::toUnderlyingType(nativeIndex)];
-        if ((nativePriority != lt::dont_download) && m_session->isFilenameExcluded(filePath.filename()))
+        if (m_session->isExcludedFileNamesEnabled()
```

Comment:
Please give it a variable outside the loop:
```c++
const bool excludedFileNamesEnabled = m_session->isExcludedFileNamesEnabled();

for (int i = 0; i < fileNames.size(); ++i)
{
    //...
    if (excludedFileNamesEnabled && (nativePriority != lt::dont_download) && m_session->isFilenameExcluded(filePath.filename()))
    //...
}
```

Reply:
Wouldn't it be simpler if `isFilenameExcluded()` will return false (by contract) if it is disabled?

Reply:
>Wouldn't it be simpler if isFilenameExcluded() will return false (by contract) if it is disabled?

Don't know if there's a cost to the other 2 comparisons but if it sees `excludedFileNamesEnabled == false` it will abort from making them. Or is it overthinking?